mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala

```diff
@@ -38,47 +42,66 @@ object MongoQueryProcessor {
 
   def apply(logicalPlan: LogicalPlan, config: Config, schemaProvided: Option[StructType] = None) = new MongoQueryProcessor(logicalPlan, config, schemaProvided)
 
-  def buildNativeQuery(requiredColums: Seq[ColumnName], filters: Array[SourceFilter]): (DBObject, DBObject) =
-    (filtersToDBObject(filters), selectFields(requiredColums))
+  def buildNativeQuery(requiredColums: Seq[ColumnName], filters: Array[SourceFilter], config: Config): (DBObject, DBObject) = {
+    (filtersToDBObject(config, filters), selectFields(requiredColums))
+  }
 
-  private def filtersToDBObject(sFilters: Array[SourceFilter]): DBObject = {
+  def filtersToDBObject(config: Config, sFilters: Array[SourceFilter], parentFilterIsNot: Boolean = false): DBObject = {
```

Comment:
Could rename the variable name?
negateAllQuery or other name... I don't know...